dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+                      Vector(fundingInput),
+                      outputs,
+                      timeouts.contractMaturity.toUInt32)
+    )
+
+    val readyToSignPSBT = psbt
+      .addSignature(remoteSig, inputIndex = 0)
+      .addUTXOToInput(fundingTx, index = 0)
+      .addScriptWitnessToInput(P2WSHWitnessV0(fundingSPK), index = 0)
+
+    val signedPSBTF = readyToSignPSBT.sign(inputIndex = 0, fundingPrivKey)
+
+    val signedCETF = signedPSBTF.flatMap { signedPSBT =>
+      val txT = signedPSBT.finalizePSBT.flatMap(_.extractTransactionAndValidate)
+      Future.fromTry(txT)
+    }
+
+    (signedCETF, P2WSHWitnessV0(toLocalSPK))
+  }
+
+  /** Constructs Remote's CET given sig*G, the funding tx's UTXOSpendingInfo and payouts */
+  def createCETRemote(
+      sigPubKey: ECPublicKey,
+      payout: CurrencyUnit,
+      remotePayout: CurrencyUnit): Future[
+    (Transaction, PartialSignature, P2WSHWitnessV0)] = {
+    val (cetLocalPrivKey, cetRemotePubKey) = if (sigPubKey == sigPubKeyWin) {
+      (cetWinPrivKey, cetRemoteWinPubKey)
+    } else {
+      (cetLosePrivKey, cetRemoteLosePubKey)
+    }
+
+    val remoteToLocalPubKey =
+      cetRemotePubKey.deriveChildPubKey(UInt32.zero).get.key
+    val localToLocalPrivKey = cetLocalPrivKey.deriveChildPrivKey(UInt32.one).key
+    val toRemotePrivKey = cetLocalPrivKey.deriveChildPrivKey(UInt32(2)).key
+
+    val pubKey = sigPubKey.add(remoteToLocalPubKey)
+
+    val toLocalSPK = P2PKWithTimeoutScriptPubKey(
+      pubKey = pubKey,
+      lockTime = ScriptNumber(timeouts.penaltyTimeout),
+      timeoutPubKey = localToLocalPrivKey.publicKey
+    )
+
+    val toLocal: TransactionOutput =
+      TransactionOutput(remotePayout + toLocalClosingFee,
+                        P2WSHWitnessSPKV0(toLocalSPK))
+    val toRemote: TransactionOutput =
+      TransactionOutput(payout, P2WPKHWitnessSPKV0(toRemotePrivKey.publicKey))
+
+    val outputs: Vector[TransactionOutput] = Vector(toLocal, toRemote)
+
+    val fundingTx = createUnsignedFundingTransaction
+    val fundingTxid = fundingTx.txIdBE
+    val fundingOutPoint = TransactionOutPoint(fundingTxid, UInt32.zero)
+    val fundingInput =
+      TransactionInput(fundingOutPoint, EmptyScriptSignature, sequence)
+
+    val unsignedTx = BaseTransaction(TransactionConstants.validLockVersion,
+                                     Vector(fundingInput),
+                                     outputs,
+                                     timeouts.contractMaturity.toUInt32)
+
+    val sigF = PSBT
+      .fromUnsignedTx(unsignedTx)
+      .addUTXOToInput(fundingTx, index = 0)
+      .addScriptWitnessToInput(P2WSHWitnessV0(fundingSPK), index = 0)
+      .sign(inputIndex = 0, fundingPrivKey)
+      .map(_.inputMaps.head.partialSignatures.head)
+
+    sigF.map((unsignedTx, _, P2WSHWitnessV0(toLocalSPK)))
+  }
+
+  lazy val createUnsignedRefundTx: Transaction = {
+    val fundingTx = createUnsignedFundingTransaction
+    val fundingTxid = fundingTx.txId
```

Comment:
So `TransactionOutPoint` has constructors for both the `DoubleSha256Digest` and `DoubleSha256DigestBE` types and handles the flipping in the `apply`, since this val is only used in an outpoint and not returned or passed anywhere, I figured it didn't matter and did the shorter one cause I'm lazy